src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -925,6 +926,71 @@ void LoadHandler::parserEndElement(GMarkupParseContext* context, const gchar* el
     }
 }
 
+void LoadHandler::fixNullPressureValues() {
+    /*
+     * Due to various bugs (see e.g. https://github.com/xournalpp/xournalpp/issues/3643), old files may contain strokes
+     * with non-positive pressure values.
+     *
+     * Those strokes thus fails the somewhat reasonable assumption that pressure values should be positive.
+     * The portions of stroke with non-positive pressure values are in any case invisible.
+     *
+     * This function fixes corrupted strokes by
+     *  1- removing every point with a non-positive pressure value.
+     *  2- if required, splitting the affected stroke into several strokes.
+     *  3- entirely deleting strokes without any valid pressure value
+     */
+    auto& pts = stroke->getPointVector();
+    if (pressureBuffer.size() >= pts.size()) {
+        // Too many pressure values. Drop the last ones
+        assert(pts.size() >= 2);  // An error has already been returned otherwise
+        pressureBuffer.resize(pts.size() - 1);
+    }
+
+    auto nextPositive = std::find_if(pressureBuffer.begin(), pressureBuffer.end(), [](double v) { return v > 0; });
+
+    std::vector<std::vector<Point>> strokePortions;
+    while (nextPositive != pressureBuffer.end()) {
+        auto nextNonPositive = std::find_if(nextPositive, pressureBuffer.end(), [](double v) { return v <= 0; });
+        size_t nValidPressureValues = static_cast<size_t>(std::distance(nextPositive, nextNonPositive));
+
+        std::vector<Point> ps;
+        ps.reserve(nValidPressureValues + 1);
+
+        std::transform(nextPositive, nextNonPositive,
+                       std::next(pts.begin(), std::distance(pressureBuffer.begin(), nextPositive)),
+                       std::back_inserter(ps), [](double v, const Point& p) { return Point(p.x, p.y, v); });
+
+        ps.emplace_back(*(pts.begin() + std::distance(pressureBuffer.begin(), nextNonPositive)));
```

Comment:
It took me some thinking to figure out why this is safe. Basically, my concern was whether we could have a stroke of size one, so that the `std::distance(...)` would evaluate to one and cause the end iterator to be dereferenced.

But this can't happen, and here's why: on the first iteration, `nextNonPositive` is guaranteed to not be the end due to the way this method is called in `parserText()`. This means that because `nextPositive` is not the end iterator (while loop condition), the stroke contains at least two points.

It follows that the indexing is safe because there must be exactly `pts.size() - 1` pressure points.

I would document this in a comment.